server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesRequest.java

```diff
@@ -61,14 +61,18 @@ public FieldCapabilitiesRequest() {}
 
     /**
      * Returns <code>true</code> iff the results should be merged.
+     *
+     * Note that when using the high-level REST client, results are always merged (this flag is always considered 'true').
      */
     boolean isMergeResults() {
         return mergeResults;
     }
 
     /**
-     * if set to <code>true</code> the response will contain only a merged view of the per index field capabilities. Otherwise only
-     * unmerged per index field capabilities are returned.
+     * If set to <code>true</code> the response will contain only a merged view of the per index field capabilities.
+     * Otherwise only unmerged per index field capabilities are returned.
+     *
+     * Note that when using the high-level REST client, results are always merged (this flag is always considered 'true').
```

Comment:
unrelated, but I am wondering why we allow to set fields either from the request body or from url parameters. I thought that one way should be enough. That also forces us to make a choice between the two in the high-level REST client. We are going for parameters now, that also means that the request object parses fields but doesn't print it out, which generally leads to no unit test coverage for the parsing code :) @jimczi do you recall why this choice was made?

Reply:
@jtibshirani would you like to address this too as a follow-up?

Reply:
Update: see https://github.com/elastic/elasticsearch/pull/30157.